tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetLineSegment.cpp

```diff
@@ -0,0 +1,276 @@
+          metavars::InterpolationTargetA>>(0, domain_creator.create_domain());
+
+  runner.simple_action<mock_interpolator<metavars, 3>,
+                       ::intrp::Actions::InitializeInterpolator<3>>(0);
+
+  const auto& box_target =
+      runner
+          .template algorithms<mock_interpolation_target<
+              metavars, metavars::InterpolationTargetA>>()
+          .at(0)
+          .template get_databox<typename mock_interpolation_target<
+              metavars, metavars::InterpolationTargetA>::initial_databox>();
+
+  const auto& box_interpolator =
+      runner.template algorithms<mock_interpolator<metavars, 3>>()
+          .at(0)
+          .template get_databox<
+              typename mock_interpolator<metavars, 3>::initial_databox>();
+
+  Slab slab(0.0, 1.0);
+  Time temporal_id(slab, 0);
+
+  runner.simple_action<
+      mock_interpolation_target<metavars, metavars::InterpolationTargetA>,
+      ::intrp::Actions::LineSegment<metavars::InterpolationTargetA, 3,
+                                    Frame::Inertial>>(0, temporal_id);
+
+  // This should not have changed.
+  CHECK(
+      db::get<::intrp::Tags::IndicesOfFilledInterpPoints>(box_target).empty());
+
+  //-:    // Should be no queued actions in mock_interpolation_target
+  //-:    CHECK(runner.is_simple_action_queue_empty<
+  //-:          mock_interpolation_target<metavars,
+  //metavars::InterpolationTargetA>>(
+  //-:        0));
+
+  // But there should be one in mock_interpolator
+  runner.invoke_queued_simple_action<mock_interpolator<metavars, 3>>(0);
+
+  //-:    // Should be no more queued actions in mock_interpolator
+  //-:    CHECK(
+  //-:        runner.is_simple_action_queue_empty<mock_interpolator<metavars,
+  //3>>(0));
+
+  const auto& vars_holders =
+      db::get<intrp::Tags::InterpolatedVarsHolders<metavars, 3>>(
+          box_interpolator);
+  const auto& vars_infos =
+      get<intrp::Vars::HolderTag<metavars::InterpolationTargetA, metavars, 3>>(
+          vars_holders)
+          .infos;
+  // Should be one entry in the vars_infos
+  CHECK(vars_infos.size() == 1);
+  const auto& info = vars_infos.at(temporal_id);
+  const auto& block_coord_holders = info.block_coord_holders;
+
+  // Should be 15 points.
+  CHECK(block_coord_holders.size() == 15);
+
+  const auto expected_block_coord_holders = [&domain_creator]() {
+    const size_t n_pts = 15;
+    tnsr::I<DataVector, 3, Frame::Inertial> points(n_pts);
+    for (size_t d = 0; d < 3; ++d) {
+      for (size_t i = 0; i < n_pts; ++i) {
+        points.get(d)[i] = 1.0 + 0.1 * i;  // Worked out by hand.
+      }
+    }
+    return block_logical_coordinates(domain_creator.create_domain(), points);
+  }();
+  for (size_t i = 0; i < 15; ++i) {
+    CHECK(block_coord_holders[i].id == expected_block_coord_holders[i].id);
+    CHECK_ITERABLE_APPROX(block_coord_holders[i].data,
+                          expected_block_coord_holders[i].data);
+  }
+}
+
```

Comment:
call a second time with another list of points, making sure you get only the new points

Reply:
Don't think I understand.

The list of points is determined from the options (read from the input file, or in this case passed in to the constructor of MockRuntimeSystem).  So I can call `::intrp::Actions::LineSegment` again, but it will compute the same list of points. But I can add them at a different temporal_id, and check that. So that's what I'll do.

BTW in real usage, ::intrp::Actions::LineSegment will be called only after the set of points has been cleared for that temporal_id.  (This will happen in an Action that you haven't seen yet, which is the only place ::intrp::Actions::LineSegment will ever be called from).  So it will never be called twice with the same temporal_id.

Reply:
I want the clear on line 62 of SendPointsToInterpolator to be triggered in the test

Reply:
That clear is always called, but the object that is cleared is never set in this PR (it will be set by an Action that I have not added yet in a PR).

Reply:
Okay, then we can defer testing this to the future PR